env0/resource_aws_oidc_credentials_test.go

```diff
@@ -0,0 +1,239 @@
+			RoleArn:  awsCredentialsResource["role_arn"].(string),
+			Duration: duration,
+		},
+		Type: client.AwsOidcCredentialsType,
+	}
+
+	updatePayload := client.AwsCredentialsCreatePayload{
+		Value: client.AwsCredentialsValuePayload{
+			RoleArn:  updatedAwsCredentialsResource["role_arn"].(string),
+			Duration: updatedDuration,
+		},
+		Type: client.AwsOidcCredentialsType,
+	}
+
+	returnValues := client.Credentials{
+		Id:             "f595c4b6-0a24-4c22-89f7-7030045de30f",
+		Name:           "test",
+		OrganizationId: "id",
+		Type:           string(client.AwsOidcCredentialsType),
+	}
+
+	otherTypeReturnValues := client.Credentials{
+		Id:             "f595c4b6-0a24-4c22-89f7-7030045de30a",
+		Name:           "test",
+		OrganizationId: "id",
+		Type:           "AWS_....",
+	}
+
+	updateReturnValues := client.Credentials{
+		Id:             returnValues.Id,
+		Name:           returnValues.Name,
+		OrganizationId: "id",
+		Type:           string(client.AwsOidcCredentialsType),
+	}
+
+	testCaseForCreateAndUpdate := resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: resourceConfigCreate(resourceType, resourceName, awsCredentialsResource),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(accessor, "name", awsCredentialsResource["name"].(string)),
+					resource.TestCheckResourceAttr(accessor, "role_arn", awsCredentialsResource["role_arn"].(string)),
+					resource.TestCheckResourceAttr(accessor, "id", returnValues.Id),
+					resource.TestCheckResourceAttr(accessor, "duration", awsCredentialsResource["duration"].(string)),
+				),
+			},
+			{
+				Config: resourceConfigCreate(resourceType, resourceName, updatedAwsCredentialsResource),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(accessor, "name", updatedAwsCredentialsResource["name"].(string)),
+					resource.TestCheckResourceAttr(accessor, "role_arn", updatedAwsCredentialsResource["role_arn"].(string)),
+					resource.TestCheckResourceAttr(accessor, "id", updateReturnValues.Id),
+					resource.TestCheckResourceAttr(accessor, "duration", updatedAwsCredentialsResource["duration"].(string)),
+				),
+			},
+		},
+	}
+
+	t.Run("create and update", func(t *testing.T) {
+		runUnitTest(t, testCaseForCreateAndUpdate, func(mock *client.MockApiClientInterface) {
+			gomock.InOrder(
+				mock.EXPECT().CredentialsCreate(&createPayload).Times(1).Return(returnValues, nil),
+				mock.EXPECT().CloudCredentials(returnValues.Id).Times(2).Return(returnValues, nil),
+				mock.EXPECT().CredentialsUpdate(returnValues.Id, &updatePayload).Times(1).Return(updateReturnValues, nil),
+				mock.EXPECT().CloudCredentials(updateReturnValues.Id).Times(1).Return(updateReturnValues, nil),
+				mock.EXPECT().CloudCredentialsDelete(returnValues.Id).Times(1).Return(nil),
+			)
+		})
+	})
+
+	t.Run("drift", func(t *testing.T) {
+		stepConfig := resourceConfigCreate(resourceType, resourceName, awsCredentialsResource)
```

Comment:
Can you please explain what is being checked here? Sounds like drift but I'm not sure how is it being checked here

Reply:
After creation, a terraform "read" is applied, but nothing is found (404 is returned).
This will force Terraform to re-create a new resource.

Reply:
Great thanks 👍